### ISSUE TITLE
[cw][sim] Ensure consistent ID for Y3 review demo process

### ISSUE THREAD

**Issue by tomjorquera** [LearnPAd/learnpad#495]
Some legwork is still required to ensure that the process ID is consistent between the CW and the SIM.
We need to do this for the new process added for final review.

This may also concerns the OR ?

**Comment by gulyx** [comment]
@tomjorquera it seems that only the SIM and the CW (i.e. (`Dashboard.xml`)[https://github.com/LearnPAd/learnpad/blob/master/lp-collaborative-workspace/lp-cw-component/lp-cw-component-ui/src/main/resources/LPUI/Dashboard.xml#L47-L50], and `SimulationEnvironment.xml`)[ https://github.com/LearnPAd/learnpad/blob/master/lp-collaborative-workspace/lp-cw-component/lp-cw-component-ui/src/main/resources/LPUI/SimulationEnvironment.xml#L49-L54]) are affected by this issue

**Comment by gulyx** [comment]
@tomjorquera I tested [`SimulationEnvironment.xml`]( https://github.com/LearnPAd/learnpad/blob/master/lp-collaborative-workspace/lp-cw-component/lp-cw-component-ui/src/main/resources/LPUI/SimulationEnvironment.xml#L49-L54) by locally removing the "hard-coded IDs", but of course it does not work. If I understood correctly form our call, likely this is because the IDs of the current version of the model provided by Barbara are not matching the IDs you are using in your BPMN file edited by hand.
Is that right? In this case, for sure we cannot guess any possible IDs the ME would generate :) and I am not sure if only adopting different hard-coded IDs ( `mod.titoloUnico`, `mod.gestioneConferenzaDiServizi`)
for the two processes will work. Probably the CW has to implement also some logic in order do decide which of the two hard-coded ID has to be passed to the SIM. What do u think?

**Comment by tomjorquera** [comment]
@gulyx: Yes the IDs have to be consistent. This is probably the reason why it did not work.

As we discussed previously, the fact that the IDs generated by the ME is a problem. I have to say that I do not know how to solve this in a satisfactory way... :worried:

Fixing the IDs and replacing them at the CW level may be the easiest to implement

Also, I am surprised that the OR is not affected by this. How does it manage to provide recommendations based on historical cases without having a consistent process ID to identify which cases actually belong to the same process ?

**Comment by gulyx** [comment]
@tomjorquera , please take a look at [this proposal](https://github.com/gulyx/learnpad/blob/fix-IDsForcing-Y2R/lp-collaborative-workspace/lp-cw-component/lp-cw-component-ui/src/main/resources/LPUI/SimulationEnvironment.xml#L60-L90). Here i changed the way the CW deal with hardcoded IDs, so that we can actually support both the scenarios (i.e. single use and multi user). Let me know.

About the values of the hard-coded IDs, we will change it later ... maybe according to some suggestion from @semmeneg (i will talk to him soon)

**Comment by gulyx** [comment]
@tomjorquera what's about this solution that relies on suffixes?
 * https://github.com/gulyx/learnpad/blob/fix-IDsForcing-Y2R/lp-collaborative-workspace/lp-cw-component/lp-cw-component-ui/src/main/resources/LPUI/SimulationEnvironment.xml#L70
 *  * https://github.com/gulyx/learnpad/blob/fix-IDsForcing-Y2R/lp-collaborative-workspace/lp-cw-component/lp-cw-component-ui/src/main/resources/LPUI/SimulationEnvironment.xml#L75

**Comment by gulyx** [comment]
@tomjorquera, @acalabro, note that with PR  #542 I included the "golden version" of the models in the repo
and I also modificed the https://github.com/gulyx/learnpad/blob/fix-IDsForcing-Y2R/lp-collaborative-workspace/lp-cw-component/lp-cw-component-ui/src/main/resources/LPUI/SimulationEnvironment.xml#L72-L78 with the correct model IDs

**Comment by gulyx** [comment]
in PR #542 I uploaded the wrong file. The correct golden version is given since #548
